server/src/client_api.rs

```diff
@@ -507,6 +514,52 @@ mod tests {
         }
     }
 
+    #[tokio::test]
+    async fn response_includes_variant_stickiness_for_strategy_variants() {
+        let features_cache: Arc<DashMap<String, ClientFeatures>> = Arc::new(DashMap::default());
+        let token_cache: Arc<DashMap<String, EdgeToken>> = Arc::new(DashMap::default());
+        let app = test::init_service(
+            App::new()
+                .app_data(Data::from(features_cache.clone()))
+                .app_data(Data::from(token_cache.clone()))
+                .service(web::scope("/api/client").service(get_features)),
+        )
+        .await;
+
+        features_cache.insert("production".into(), cached_client_features());
+        let mut production_token = EdgeToken::try_from(
+            "*:production.03fa5f506428fe80ed5640c351c7232e38940814d2923b08f5c05fa7".to_string(),
+        )
+        .unwrap();
+        production_token.token_type = Some(TokenType::Client);
+        production_token.status = TokenValidationStatus::Validated;
+        token_cache.insert(production_token.token.clone(), production_token.clone());
+        let req = make_features_request_with_token(production_token.clone()).await;
+        let res: ClientFeatures = test::call_and_read_body_json(&app, req).await;
+
+        assert_eq!(res.features.len(), cached_client_features().features.len());
+        let strategy_variant_stickiness = res
+            .features
```

Comment:
@chriswk I hate this chain so much but I can't spot an obvious way accurately assert the stickiness is Some without rooting around in the data structure. I'm very open to suggestions!

Reply:
I'd say bring some lens library. ;) in all seriousness though, this is terrible, but consequence of having a deep data structure I guess.

Reply:
Oof I really don't want to take on a new dev dependency for a single test when I don't see us getting much out of it for other things. I can live with this (with a small amount of crying maybe)